layouts/Error.tsx

```diff
@@ -0,0 +1,79 @@
+// CSS
+import styles from '../styles/layouts/Error.module.scss';
+// Components
+import NavBar from '../components/NavBar';
+import Button, { ButtonType } from '../components/Button';
+import React from 'react';
+import { useEffect, useState } from 'react';
+import Footer from '@/components/Footer';
+
+const departure_url = 'https://tcsca';
+
+export default function ErrorLayout({
+  children,
+}:{
+  children: React.ReactNode
+}) {
+  const [allowReport, setAllowReport] = useState(false);
+  useEffect(() => {
+    // If the user is coming from a broken link on our site, they can report an issue
```

Comment:
Can you just add a comment below this, stating `Note that the Document. referrer property is not always reliable`.

I think the way this is implemented is the best option because we are a website most of the errors will be real 404s from mistypes, but we care about ones internally. but document refferer can be blocked by extensions, browsers and I think devices so its not 100% reliable.